### ISSUE TITLE
Make revealing debug console on debug start a configuration

### ISSUE THREAD

**Issue by v-pavanp** [Microsoft/vscode#5889]
- VSCode Version: 1.1.0-alpha
- OS Version: WInodws10

Steps to Reproduce:
1. Launch app.
2. Set configuration for debugging : task.json, launch.json and tsconfig.json.
3. Set "Showoutput" property 'always' to get message in the output window.
4. Start debugging and verify debug window.
6. Now perform build operation with CTRL+Shift+B and verify output window.
7. Start debugging again and verify debug console window.

Actual: Debug console window is not showing up when user start debugging after build.
Expected: Debug console window should be on top when user start debugging.

![outoutwindowissue](https://cloud.githubusercontent.com/assets/17735796/14851037/012047bc-0c32-11e6-8167-ef728e848126.gif)

**Comment by bpasero** [comment]
I think we got smarter here about when to show the debug console. @isidorn please chime in

**Comment by isidorn** [comment]
We only reveal the debug console on the first debug session start, not on consecutive ones.
We are no longer forcing the debug console on every debug start because that was too obtrusive for some debug users.
Closing as designed as we would like to get feedback from users for this change. If users prefere that we always show the debug console, then we can easily go back to our old behavior, but for now I would really like to get more feeback.

**Comment by v-pavanp** [comment]
Can we make it configurable?

**Comment by isidorn** [comment]
Yes, that's a good idea. Let me reopen this and rename the issue.

**Comment by isidorn** [comment]
@bpasero how does `debug.openConsoleOnStart` sound to you? Default value would be `false`

**Comment by bpasero** [comment]
@isidorn maybe this should be a setting within a launch config instead of global?

**Comment by isidorn** [comment]
@bpasero also fine by me. I think that actually makes more sense since you might want different behavior per configuration. Like in your case, launching unit tests vs launching app. If we put it inside `launch.json` I would just call it `openConsoleOnStart`

@weinand opinions?

**Comment by weinand** [comment]
@isidorn some comments:
- since this will only apply to the internal console, we probably should use 'internal' in the attribute name.
- instead of making the attribute boolean, I suggest to make it more open ended like Dirk's `showOutput`. It is very likely that there will be more settings in the future. I suggest to call the attribute `internalConsoleOptions` and use a value of `openOnSessionStart` for the case considered here.

**Comment by bpasero** [comment]
Oh yeah, I was actually thinking about how it is possible for tasks to control this and was expecting something similar.

**Comment by isidorn** [comment]
I applied the changes you proposed and explained them in #4045

**Comment by v-pavanp** [comment]
Thanks for the updates, I verified and getting debug console window all the time with my setting. But I observed that application behavior got changed for task window. I used to get task window all the time whenever I performed CTRL+SHIFT+B (build command) but now task window is not displaying all the time. With the latest alpha build I observed that user needs to explicitly open task window to see the build output. Did we also make the task window configurable? If so what are the settings I can use to get it all the time for build action.
Basically my intention here is that console windows (task and debug) should be interchangeable with user's action and of course according to local setting.
